### PR TITLE
ISPN-13971 Exclude new NativeTransport

### PR DESCRIPTION
**Pull request by tristantarrant**
Required after https://github.com/infinispan/infinispan/pull/10185

### CODE REVIEW

**Comment by ryanemerson** [comment]
There's an issue with the substitutions on CI:

```
Error: Substitution target for org.infinispan.quarkus.server.runtime.graal.Delete_org_infinispan_client_hotrod_impl_transport_netty_NativeTransport is not loaded. Use field `onlyWith` in the `TargetClass` annotation to make substitution only active when needed.

com.oracle.svm.core.util.UserError$UserException: Substitution target for org.infinispan.quarkus.server.runtime.graal.Delete_org_infinispan_client_hotrod_impl_transport_netty_NativeTransport is not loaded. Use field `onlyWith` in the `TargetClass` annotation to make substitution only active when needed.

	at com.oracle.svm.core.util.UserError.abort(UserError.java:72)

	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.findTargetClass(AnnotationSubstitutionProcessor.java:969)

	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.handleClass(AnnotationSubstitutionProcessor.java:301)

	at com.oracle.svm.hosted.substitute.AnnotationSubstitutionProcessor.init(AnnotationSubstitutionProcessor.java:279)

	at com.oracle.svm.hosted.NativeImageGenerator.createAnnotationSubstitutionProcessor(NativeImageGenerator.java:933)

	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:846)

	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:555)

	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:515)

	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)

	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)

	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)

	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:615)
```

**Comment by tristantarrant** [comment]
@ryanemerson fun thing: the original file was wrong because it mentions "client" whereas it was still substituting the server version...
I'll remove it completely

File: server/runtime/src/main/java/org/infinispan/quarkus/server/runtime/graal/SubstituteNettyClasses.java [hunk 1]
```diff
@@ -51,6 +51,10 @@ public static MultithreadEventLoopGroup buildEventLoop(int nThreads, ThreadFacto
    }
 }
 
+/*
+ * Reinstate once we include the Hot Rod client
```
Comment (ryanemerson) [inline server/runtime/src/main/java/org/infinispan/quarkus/server/runtime/graal/SubstituteNettyClasses.java:55]: Why do we want to include the client in the server?

**Review by ryanemerson — COMMENTED** [review]
The native build is still failing.

File: server/runtime/src/main/java/org/infinispan/quarkus/server/runtime/graal/SubstituteNettyClasses.java (hunk 1 above)
Comment (tristantarrant) [reply server/runtime/src/main/java/org/infinispan/quarkus/server/runtime/graal/SubstituteNettyClasses.java:55]: Remote Cache Store :)

**Comment by ryanemerson** [comment]
Closing as superseded by #99

**Pull request closed without merging**